pkg/plugins/policies/meshretry/plugin/xds/configurer.go

```diff
@@ -165,6 +177,53 @@ func configureRateLimitedRetryBackOff(rateLimitedBackOff *api.RateLimitedBackOff
 	return rateLimitedRetryBackoff
 }
 
+func configureHostSelectionPredicates(hostSelection *[]api.Predicate, policy *envoy_route.RetryPolicy) {
+	var retryHostPredicates []*envoy_route.RetryPolicy_RetryHostPredicate
+	for _, hostSelect := range pointer.Deref(hostSelection) {
+		switch hostSelect.PredicateType {
+		case api.OmitPreviousHosts:
+			prevHosts, _ := util_proto.MarshalAnyDeterministic(
+				&envoy_prev_hosts.PreviousHostsPredicate{},
+			)
+			retryHostPredicates = append(retryHostPredicates,
+				&envoy_route.RetryPolicy_RetryHostPredicate{
+					Name: "envoy.retry_host_predicates.previous_hosts",
+					ConfigType: &envoy_route.RetryPolicy_RetryHostPredicate_TypedConfig{
+						TypedConfig: prevHosts,
+					},
+				},
+			)
+		case api.OmitHostsWithTags:
+			taggedHosts, _ := util_proto.MarshalAnyDeterministic(
```

Comment:
we shouldn't swallow errors (applies to 3 cases), either propagate that up or use `MustMarshalAny` if we're pretty sure it will never fail

Reply:
Propagated them up